editor.planx.uk/src/@planx/components/Checklist/Public/Public.tsx

```diff
@@ -90,18 +95,36 @@ const VisibleChecklist: React.FC<Props> = (props) => {
   const layout = getLayout({ options, groupedOptions });
   const flatOptions = getFlatOptions({ options, groupedOptions });
 
-  const changeCheckbox = (id: string) => (_checked: any) => {
-    const newCheckedIds = formik.values.checked.includes(id)
-      ? formik.values.checked.filter((x) => x !== id)
-      : [...formik.values.checked, id];
-
-    formik.setFieldValue(
-      "checked",
-      newCheckedIds.sort((a, b) => {
-        const originalIds = flatOptions.map((cb) => cb.id);
-        return originalIds.indexOf(a) - originalIds.indexOf(b);
-      }),
-    );
+  const sortCheckedIds = (ids: string[]): string[] => {
+    const originalIds = flatOptions.map((cb) => cb.id);
+    return ids.sort((a, b) => originalIds.indexOf(a) - originalIds.indexOf(b));
+  };
+
+  const exclusiveOrOption =
+    options?.find((option) => option.data?.exclusive === true) || undefined;
+
+  const exclusiveOptionIsChecked =
+    exclusiveOrOption && formik.values.checked.includes(exclusiveOrOption.id);
+
+  const changeCheckbox = (id: string) => () => {
```

Comment:
Thanks Daf, I realised one of the permutations was unnecessary, so that cut one of the branches out, and makes it a lot easier already. I've also extracted more of these code blocks out into functions e.g. `toggleExclusiveOption` and `setCheckedFieldValue`.

In general I try to avoid comments - I believe that comments can mostly be made unnecessary if the code is written clearly enough!

Reply:
This is a good instinct to have re:comments 👍 

This functions reads much better now thanks.